src/index.ts

```diff
@@ -1,55 +1,141 @@
-import { existsSync, readFileSync } from 'fs';
+import { UploadSourceMapsHandler } from '@byteboost/cli/sdk';
+import { readFileSync } from 'fs';
 import path from 'path';
-import mime from 'mime';
-import FormData from 'form-data';
-import fetch from 'node-fetch';
+import type Webpack from 'webpack';
 
-import Webpack from 'webpack';
+export interface OptionsInterface {
+  organization: string;
+  domain: string;
+  token: string;
+  logging?: boolean;
+  versionFromPackageJSON?: boolean;
+  version?: string;
+  debug?: boolean;
+}
 
 export class ByteboostSourcemaps {
-  apply(compiler: Webpack.Compiler) {
-    compiler.hooks.done.tap('ByteboostSourcemaps', (stats) => {
-      stats.compilation.chunks.forEach((chunk) => {
-        chunk.files.forEach(async (file) => {
-          const codePath = path.join(compiler.options.output.path!, file);
-          const sourcemapPath = path.join(
-            compiler.options.output.path!,
-            `${file}.map`,
-          );
-
-          if (!existsSync(codePath)) {
-            console.warn(`Code file not found: ${codePath}`);
-            return;
-          }
-
-          if (!existsSync(sourcemapPath)) {
-            console.warn(`Sourcemap file not found: ${sourcemapPath}`);
-            return;
-          }
-
-          const form = new FormData();
-
-          form.append('files', readFileSync(codePath), {
-            filename: file,
-            contentType: mime.getType(codePath) ?? undefined,
-          });
-
-          form.append('files', readFileSync(sourcemapPath), {
-            filename: `${file}.map`,
-            contentType: mime.getType(sourcemapPath) ?? undefined,
-          });
-
-          await fetch(`https://dev.api.byteboost.com/sourcemap`, {
-            method: 'POST',
-            headers: {
-              ...form.getHeaders(),
-              ['X-Auth-Token']: 'test',
-            },
-            insecureHTTPParser: true,
-            body: form,
-          });
-        });
-      });
+  static defaultOptions: OptionsInterface = {
+    organization: '',
+    domain: '',
+    token: '',
+    logging: false,
+    debug: false,
+  };
+
+  public options: Partial<OptionsInterface> = {};
+
+  private compiler: Webpack.Compiler | null = null;
+
+  constructor(options: Partial<OptionsInterface> = {}) {
+    if (!options.token) {
+      throw new Error('token is required');
+    }
+
+    if (!options.organization) {
+      throw new Error('organization is required');
+    }
+
+    if (!options.domain) {
+      throw new Error('domain is required');
+    }
+
+    if (!options.version && !options.versionFromPackageJSON) {
+      throw new Error('version is required');
+    }
+
+    if (options.version && options.versionFromPackageJSON) {
+      throw new Error(
+        'version and versionFromPackageJSON cannot be used together',
+      );
+    }
+
```

Comment:
Should display that these error are from the byteboost plugin

e..g prefix with byteboost or something